From ecafd75954ae61d1ec574b44b7e5da3c60be6131 Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Mon, 19 Sep 2022 18:50:46 +0200
Subject: [PATCH] [PATCH] Drop AnimationTimeline, adjust IDL terms in Web
 Animations

Some terms are partially re-defined in Level 2. Parts of the patch that update
these definitions are only needed because Level 2 is a delta spec. They will
likely need to be kept around for as long as that remains the case.
---
 ed/idl/web-animations.idl | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/ed/idl/web-animations.idl b/ed/idl/web-animations.idl
index 1b078e6ea..6030dad33 100644
--- a/ed/idl/web-animations.idl
+++ b/ed/idl/web-animations.idl
@@ -24,8 +19,6 @@ interface Animation : EventTarget {
              attribute DOMString                id;
              attribute AnimationEffect?         effect;
              attribute AnimationTimeline?       timeline;
-             attribute double?                  startTime;
-             attribute double?                  currentTime;
              attribute double                   playbackRate;
     readonly attribute AnimationPlayState       playState;
     readonly attribute AnimationReplaceState    replaceState;
@@ -58,12 +51,9 @@ interface AnimationEffect {
 };
 
 dictionary EffectTiming {
-    double                             delay = 0;
-    double                             endDelay = 0;
     FillMode                           fill = "auto";
     double                             iterationStart = 0.0;
     unrestricted double                iterations = 1.0;
-    (unrestricted double or DOMString) duration = "auto";
     PlaybackDirection                  direction = "normal";
     DOMString                          easing = "linear";
 };
@@ -84,9 +74,6 @@ enum FillMode { "none", "forwards", "backwards", "both", "auto" };
 enum PlaybackDirection { "normal", "reverse", "alternate", "alternate-reverse" };
 
 dictionary ComputedEffectTiming : EffectTiming {
-    unrestricted double  endTime;
-    unrestricted double  activeDuration;
-    double?              localTime;
     double?              progress;
     unrestricted double? currentIteration;
 };
@@ -156,14 +143,3 @@ partial interface mixin DocumentOrShadowRoot {
 };
 
 Element includes Animatable;
-
-[Exposed=Window]
-interface AnimationPlaybackEvent : Event {
-    constructor(DOMString type, optional AnimationPlaybackEventInit eventInitDict = {});
-    readonly attribute double? currentTime;
-    readonly attribute double? timelineTime;
-};
-dictionary AnimationPlaybackEventInit : EventInit {
-    double? currentTime = null;
-    double? timelineTime = null;
-};
-- 
2.36.0.windows.1

